File: android/lca/src/main/java/com/dreampany/lca/data/model/Exchange.java
/*
package com.dreampany.lca.data.model;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.Index;

import android.annotation.SuppressLint;
import android.os.Parcel;
import androidx.annotation.NonNull;

import com.dreampany.frame.data.model.Base;
import com.dreampany.lca.misc.Constants;

*/
/**
 * Created by Hawladar Roman on 24/6/18.
 * Dreampany Ltd
 * dreampanymail@gmail.com
 *//*

@Entity(indices = {@Index(value = {Constants.Exchange.ID}, unique = true)},
        primaryKeys = {Constants.Exchange.ID})
public class Exchange extends Base {

    private String exchange;
    private String fromSymbol;
    private String toSymbol;
    private double volume24h;
    private double volume24hTo;

    @Ignore
    public Exchange() {

    }

    public Exchange(String id) {
        this.id = id;
    }


    @Ignore
    private Exchange(Parcel in) {
        super(in);
        exchange = in.readString();
        fromSymbol = in.readString();
        toSymbol = in.readString();
        volume24h = in.readDouble();
        volume24hTo = in.readDouble();
    }

    @Override
    public void writeToParcel(@NonNull Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
        dest.writeString(exchange);
        dest.writeString(fromSymbol);
        dest.writeString(toSymbol);
        dest.writeDouble(volume24h);
        dest.writeDouble(volume24hTo);
    }

    public static final Creator<Exchange> CREATOR = new Creator<Exchange>() {
        @Override
        public Exchange createFromParcel(Parcel in) {
            return new Exchange(in);
        }

        @Override
        public Exchange[] newArray(int size) {
            return new Exchange[size];
        }
    };

    @Override
    public String toString() {
        return "Exchange{" +
                "exchange=" + exchange +
                ", toSymbol='" + toSymbol + '\'' +
                ", fromSymbol='" + fromSymbol + '\'' +
                '}';
    }

    public void setExchange(String exchange) {
        this.exchange = exchange;
    }

    public void setFromSymbol( String fromSymbol) {
        this.fromSymbol = fromSymbol;
    }

    public void setToSymbol( String toSymbol) {
        this.toSymbol = toSymbol;
    }

    public void setVolume24h(double volume24h) {
        this.volume24h = volume24h;
    }

    public void setVolume24hTo(double volume24hTo) {
        this.volume24hTo = volume24hTo;
    }

    public String getExchange() {
        return exchange;
    }

    public String getFromSymbol() {
        return fromSymbol;
    }

    public String getToSymbol() {
        return toSymbol;
    }

    public double getVolume24h() {
        return volume24h;
    }

    public double getVolume24hTo() {
        return volume24hTo;
    }
}
*/

File: android/lca/src/main/kotlin/com/dreampany/lca/data/model/Exchange.kt
package com.dreampany.lca.data.model

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.Index
import com.dreampany.framework.data.model.Base
import com.dreampany.framework.util.TimeUtilKt
import com.dreampany.lca.misc.Constants
import kotlinx.android.parcel.Parcelize

/**
 * Created by Roman-372 on 8/2/2019
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
@Parcelize
@Entity(
    indices = [Index(
        value = [Constants.Exchange.ID],
        unique = true
    )],
    primaryKeys = [Constants.Exchange.ID]
)
data class Exchange (
    override var time: Long = Constants.Default.LONG,
    override var id: String = Constants.Default.STRING,
    var exchange: String = Constants.Default.STRING,
    var fromSymbol: String = Constants.Default.STRING,
    var toSymbol: String = Constants.Default.STRING,
    var volume24h: Double = Constants.Default.DOUBLE,
    var volume24hTo: Double = Constants.Default.DOUBLE
) : Base() {

    @Ignore
    constructor() : this(time = TimeUtilKt.currentMillis()) {}
    constructor(id: String) : this(time = TimeUtilKt.currentMillis(), id = id) {}

    override fun toString(): String {
        return "Exchange{" +
                "exchange=" + exchange +
                ", toSymbol='" + toSymbol + '\''.toString() +
                ", fromSymbol='" + fromSymbol + '\''.toString() +
                '}'.toString()
    }
}